File: tasks/2023/online3/text.cpp
// Full score solution
// @check-accepted: *

#include <iostream>
#include <array>
#include <vector>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    int n, k;
    cin >> n >> k;
    vector<string> t(n);
    vector<array<int, 26>> cnt(k);
    for (string& s : t) {
        cin >> s;
        for (int i = 0; i < k; i++)
            cnt[i][s[i] -'a']++;
    }

    int mind = n * k;
    int ans = 0;
    for (int i = 0; i < n; i++) {
        int d = 0;
        for (int j = 0; j < k; j++) {
            d += n - cnt[j][t[i][j] - 'a'];
        }
        if (d < mind) {
            mind = d;
            ans = i;
        }
    }
    
    cout << ans << endl;
    return 0;
}

File: tasks/2023/online3/text.py
#!/usr/bin/env python3
# full score
# @check-accepted: *

n, k = map(int, input().strip().split())
t = [input().strip() for i in range(n)]

cnt = [[0]*26 for i in range(k)]
for s in t:
    for i in range(k):
        cnt[i][ord(s[i]) - ord('a')] += 1

ans = 0
mind = n * k
for i in range(n):
    d = 0
    for j in range(k):
        d += n - cnt[j][ord(t[i][j]) - ord('a')]
    if d < mind:
        mind = d
        ans = i

print(ans)